src/main/java/io/confluent/connect/elasticsearch/bulk/BulkProcessor.java

```diff
@@ -188,15 +191,21 @@ Runnable farmerTask() {
   private synchronized Future<BulkResponse> submitBatch() {
     final int numUnsentRecords = unsentRecords.size();
     assert numUnsentRecords > 0;
-    final int batchableSize = Math.min(batchSize, numUnsentRecords);
-    final List<R> batch = new ArrayList<>(batchableSize);
-    for (int i = 0; i < batchableSize; i++) {
+    final List<R> batch = new ArrayList<>();
+    long batchSizeBytes = 0;
+    int countBatchRecords = 0;
+    R record = unsentRecords.getFirst();
+    while (record != null && (batchSizeBytes + record.toString().length()) <= maxBatchSizeBytes
+        && countBatchRecords + 1 <= batchSize) {
       batch.add(unsentRecords.removeFirst());
+      batchSizeBytes += record.toString().length();
```

Comment:
```suggestion
    R record = unsentRecords.getFirst();
    int recordSize = record.toString().length()
    while (record != null && (batchSizeBytes + recordSize) <= maxBatchSizeBytes
        && countBatchRecords + 1 <= batchSize) {
      batch.add(unsentRecords.removeFirst());
      batchSizeBytes += recordSize);
```
We can skip performing this operation twice.